test/helpers/module.py

```diff
@@ -44,13 +42,13 @@ def module_reload(filename):
 
     """
 
-    module_name = 'apprise.plugins.{}'.format(
-        re.match(r'^(.+)(\.py)?$', os.path.basename(filename), re.I).group(1))
+    module_name = f"apprise.plugins.{name}"
 
     reload(sys.modules['apprise.common'])
     reload(sys.modules['apprise.attachment'])
     reload(sys.modules['apprise.config'])
-    reload(sys.modules[module_name])
+    if module_name in sys.modules:
+        reload(sys.modules[module_name])
```

Comment:
> I wonder if this whole time nothing works right is because we aren't also updating the `globals()`?

I am still not following. I think everything "works right" in its own sense. No problem here, until Python 3.11 arrived.

This revealed a flaw with the module vs. class reference mixup, and this patch will get things straight in this regard. After that, it will be way easier to reason about the code for humans, and it will be ready running on Python 3.11 and newer.